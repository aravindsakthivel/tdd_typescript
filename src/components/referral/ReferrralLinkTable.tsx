import { FC } from 'react';
import { ReferralRowTemplate } from './ReferralRowTemplate';
import { Table } from 'antd';
import './referralLinkTable.css';

interface ReferralLinkTableDataInterface {
  key?: string;
  urlLink: string;
  createdOn: string;
  destination: string;
  onboarded: string;
}

const ReferrralLinkTable: FC = () => {
  let referralInfoList: ReferralLinkTableDataInterface[] = [
    {
      key: '1',
      urlLink: 'https://mpl.live/mycustomcodeforamqzingoffers ',
      createdOn: '12:01pm, 12 Mar 2020',
      destination: '1 days',
      onboarded: '12',
    },
    {
      key: '2',
      urlLink: 'https://mpl.live/mycustomcodeforamqzingoffers ',
      createdOn: '12:01pm, 12 Mar 2020',
      destination: '3 days',
      onboarded: '33',
    },
    {
      key: '3',
      urlLink: 'https://mpl.live/mycustomcodeforamqzingoffers ',
      createdOn: '12:01pm, 12 Mar 2020',
      destination: '7 days',
      onboarded: '325',
    },
    {
      key: '4',
      urlLink: 'https://mpl.live/mycustomcodeforamqzingoffers ',
      createdOn: '12:01pm, 12 Mar 2020',
      destination: '8 days',
      onboarded: '5466',
    },
    {
      key: '5',
      urlLink: 'https://mpl.live/mycustomcodeforamqzingoffers ',
      createdOn: '12:01pm, 12 Mar 2020',
      destination: '7 days',
      onboarded: '325',
    },
  ];

  const columns = [
    {
      title: 'URL / Link',
      dataIndex: 'urlLink',
      key: 'urlLink',
    },
    {
      title: 'Created On',
      dataIndex: 'createdOn',
      key: 'createdOn',
    },
    {
      title: 'Destination',
      dataIndex: 'destination',
      key: 'destination',
    },
    {
      title: 'Onboarded',
      dataIndex: 'onboarded',
      key: 'onboarded',
    },
  ];
  return (
    <ReferralRowTemplate title={'History'} last={true}>
      <Table
        className="referral-link-table"
        columns={columns}
        dataSource={referralInfoList}
        style={{ backgroundColor: '#391958' }}
        pagination={referralInfoList.length > 5 ? { pageSize: 5 } : false}
      />
    </ReferralRowTemplate>
  );
};

export { ReferrralLinkTable };
