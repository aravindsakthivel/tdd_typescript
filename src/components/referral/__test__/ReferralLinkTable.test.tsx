import ReactDOM from 'react-dom';
import { render, cleanup, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { mockConfig } from '../../mock';
import { ReferrralLinkTable } from '../ReferrralLinkTable';

afterEach(cleanup);
beforeEach(() => {
  mockConfig();
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ReferrralLinkTable />, div);
});

it('check if table head title presents', async () => {
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  const data = render(<ReferrralLinkTable />);
  expect(await screen.findByText('URL / Link')).toBeInTheDocument();
  expect(await screen.findByText('Created On')).toBeInTheDocument();
  expect(await screen.findByText('Destination')).toBeInTheDocument();
  expect(await screen.findByText('Onboarded')).toBeInTheDocument();
});

it('matches snapshot', () => {
  const tree = renderer.create(<ReferrralLinkTable />).toJSON();
  expect(tree).toMatchSnapshot();
});
