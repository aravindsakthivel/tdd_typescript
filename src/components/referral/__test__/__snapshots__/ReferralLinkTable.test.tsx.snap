// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="ant-row referral-row-template-class general-background-border no-border-last"
  style={Object {}}
>
  <div
    className="ant-col ant-col-24"
    style={Object {}}
  >
    <h5
      className="ant-typography"
      data-testid="referral-row-title"
      style={
        Object {
          "WebkitLineClamp": undefined,
        }
      }
    >
      History
    </h5>
  </div>
  <div
    className="ant-col ant-col-23"
    data-testid="referral-row-children"
    style={Object {}}
  >
    <div
      className="ant-table-wrapper referral-link-table"
      style={
        Object {
          "backgroundColor": "#391958",
        }
      }
    >
      <div
        className="ant-spin-nested-loading"
      >
        <div
          className="ant-spin-container"
        >
          <div
            className="ant-table"
          >
            <div
              className="ant-table-container"
            >
              <div
                className="ant-table-content"
                onScroll={[Function]}
                style={Object {}}
              >
                <table
                  style={
                    Object {
                      "tableLayout": "auto",
                    }
                  }
                >
                  <colgroup />
                  <thead
                    className="ant-table-thead"
                  >
                    <tr>
                      <th
                        className="ant-table-cell"
                        colSpan={null}
                        rowSpan={null}
                        style={Object {}}
                      >
                        URL / Link
                      </th>
                      <th
                        className="ant-table-cell"
                        colSpan={null}
                        rowSpan={null}
                        style={Object {}}
                      >
                        Created On
                      </th>
                      <th
                        className="ant-table-cell"
                        colSpan={null}
                        rowSpan={null}
                        style={Object {}}
                      >
                        Destination
                      </th>
                      <th
                        className="ant-table-cell"
                        colSpan={null}
                        rowSpan={null}
                        style={Object {}}
                      >
                        Onboarded
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    className="ant-table-tbody"
                  >
                    <tr
                      className="ant-table-row ant-table-row-level-0"
                      data-row-key="1"
                      onClick={[Function]}
                      style={Object {}}
                    >
                      <td
                        className="ant-table-cell"
                        colSpan={null}
                        rowSpan={null}
                        style={Object {}}
                      >
                        https://mpl.live/mycustomcodeforamqzingoffers 
                      </td>
                      <td
                        className="ant-table-cell"
                        colSpan={null}
                        rowSpan={null}
                        style={Object {}}
                      >
                        12:01pm, 12 Mar 2020
                      </td>
                      <td
                        className="ant-table-cell"
                        colSpan={null}
                        rowSpan={null}
                        style={Object {}}
                      >
                        1 days
                      </td>
                      <td
                        className="ant-table-cell"
                        colSpan={null}
                        rowSpan={null}
                        style={Object {}}
                      >
                        12
                      </td>
                    </tr>
                    <tr
                      className="ant-table-row ant-table-row-level-0"
                      data-row-key="2"
                      onClick={[Function]}
                      style={Object {}}
                    >
                      <td
                        className="ant-table-cell"
                        colSpan={null}
                        rowSpan={null}
                        style={Object {}}
                      >
                        https://mpl.live/mycustomcodeforamqzingoffers 
                      </td>
                      <td
                        className="ant-table-cell"
                        colSpan={null}
                        rowSpan={null}
                        style={Object {}}
                      >
                        12:01pm, 12 Mar 2020
                      </td>
                      <td
                        className="ant-table-cell"
                        colSpan={null}
                        rowSpan={null}
                        style={Object {}}
                      >
                        3 days
                      </td>
                      <td
                        className="ant-table-cell"
                        colSpan={null}
                        rowSpan={null}
                        style={Object {}}
                      >
                        33
                      </td>
                    </tr>
                    <tr
                      className="ant-table-row ant-table-row-level-0"
                      data-row-key="3"
                      onClick={[Function]}
                      style={Object {}}
                    >
                      <td
                        className="ant-table-cell"
                        colSpan={null}
                        rowSpan={null}
                        style={Object {}}
                      >
                        https://mpl.live/mycustomcodeforamqzingoffers 
                      </td>
                      <td
                        className="ant-table-cell"
                        colSpan={null}
                        rowSpan={null}
                        style={Object {}}
                      >
                        12:01pm, 12 Mar 2020
                      </td>
                      <td
                        className="ant-table-cell"
                        colSpan={null}
                        rowSpan={null}
                        style={Object {}}
                      >
                        7 days
                      </td>
                      <td
                        className="ant-table-cell"
                        colSpan={null}
                        rowSpan={null}
                        style={Object {}}
                      >
                        325
                      </td>
                    </tr>
                    <tr
                      className="ant-table-row ant-table-row-level-0"
                      data-row-key="4"
                      onClick={[Function]}
                      style={Object {}}
                    >
                      <td
                        className="ant-table-cell"
                        colSpan={null}
                        rowSpan={null}
                        style={Object {}}
                      >
                        https://mpl.live/mycustomcodeforamqzingoffers 
                      </td>
                      <td
                        className="ant-table-cell"
                        colSpan={null}
                        rowSpan={null}
                        style={Object {}}
                      >
                        12:01pm, 12 Mar 2020
                      </td>
                      <td
                        className="ant-table-cell"
                        colSpan={null}
                        rowSpan={null}
                        style={Object {}}
                      >
                        8 days
                      </td>
                      <td
                        className="ant-table-cell"
                        colSpan={null}
                        rowSpan={null}
                        style={Object {}}
                      >
                        5466
                      </td>
                    </tr>
                    <tr
                      className="ant-table-row ant-table-row-level-0"
                      data-row-key="5"
                      onClick={[Function]}
                      style={Object {}}
                    >
                      <td
                        className="ant-table-cell"
                        colSpan={null}
                        rowSpan={null}
                        style={Object {}}
                      >
                        https://mpl.live/mycustomcodeforamqzingoffers 
                      </td>
                      <td
                        className="ant-table-cell"
                        colSpan={null}
                        rowSpan={null}
                        style={Object {}}
                      >
                        12:01pm, 12 Mar 2020
                      </td>
                      <td
                        className="ant-table-cell"
                        colSpan={null}
                        rowSpan={null}
                        style={Object {}}
                      >
                        7 days
                      </td>
                      <td
                        className="ant-table-cell"
                        colSpan={null}
                        rowSpan={null}
                        style={Object {}}
                      >
                        325
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
